Source/missiles.cpp

```diff
@@ -1953,20 +1941,20 @@ void AddMagmaball(MissileStruct &missile, Point dst, Direction /*midir*/)
 void AddTeleport(MissileStruct &missile, Point dst, Direction /*midir*/)
 {
 	missile._miDelFlag = true;
+
 	for (int i = 0; i < 6; i++) {
 		int k = CrawlNum[i];
 		int ck = k + 2;
 		for (auto j = static_cast<uint8_t>(CrawlTable[k]); j > 0; j--, ck += 2) {
-			int tx = dst.x + CrawlTable[ck - 1];
-			int ty = dst.y + CrawlTable[ck];
-			if (0 < tx && tx < MAXDUNX && 0 < ty && ty < MAXDUNY) {
-				if (IsTileNotSolid({ tx, ty }) && dMonster[tx][ty] == 0 && dObject[tx][ty] == 0 && dPlayer[tx][ty] == 0) {
-					missile.position.tile = { tx, ty };
-					missile.position.start = { tx, ty };
-					missile._miDelFlag = false;
-					i = 6;
-					break;
-				}
+			Point target = dst + Displacement { CrawlTable[ck - 1], CrawlTable[ck] };
+			if (!InDungeonBounds(target) || !IsTileNotSolid(target))

```

Comment:
```suggestion
			if (!InDungeonBounds(target) || IsTileSolid(target))
```
The double negative irks me a bit 😂. While `IsTileSolid != !IsTileNotSolid` the only exceptional case is when the point is out of bounds which gets checked first, so this should be fine.